authenticators/cf_authenticator.go

```diff
@@ -83,7 +84,16 @@ func (cfa *CFAuthenticator) Authenticate(metadata ssh.ConnMetadata, password []b
 		return nil, err
 	}
 
-	processGuid, err := cfa.checkAccess(logger, appGuid, index, string(cred))
+	parts := strings.Split(cred, " ")
+	tokenString := parts[1]
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		return []byte("Doesntmatter"), nil
+	})
+
+	username := token.Claims["user_name"].(string)
```

Comment:
it MUST be a string because the token is returned from exchangeAccessCodeForToken and it builds up a string of 2 pieces and returns a longer string.  Then I split the string in several pieces.  The parse could in theory return something strange here but most likely not.  I'll put a check in and if it fails I will just use "unknown" as the username and principal.